### DISCUSSION TITLE
How to set `cmake.cmakePath` for an SDK in devcontainer?

### DISCUSSION THREAD

**Discussion by skycaptain** [microsoft/vscode-cmake-tools#4060] [General]
I'm setting up a development environment using devcontainers for a project that uses a cross-compiling SDK.

Our setup defines a path to the native sysroot installation using an environment variable called `SDK_NATIVE_SYSROOT`. We aim to use this variable to set `cmake.cmakePath`. The documentation states that we can use environment variables in the configuration like this: `${env:VARIABLE_NAME}`.

Here's what I've tried in my configuration file:

```
{
  "image": "[redacted]",
  "customizations": {
    "vscode": {
      "extensions": [
        "ms-vscode.cpptools-extension-pack"
      ],
      "settings": {
        "cmake.cmakePath": "${env:SDK_NATIVE_SYSROOT}/usr/bin/cmake"
      }
    }
  }
}
```

However, this doesn't work. When I run it, I get an error saying it can't find CMake at `/usr/bin/cmake`. It seems like `${env:SDK_NATIVE_SYSROOT}` is being replaced with an empty string.

Interestingly, if I put the same setting in `.vscode/settings.json`, it works fine.

It appears that settings in the devcontainer.json file are evaluated using the `localEnv` before being sent to the devcontainer, while settings in `.vscode/settings.json` are evaluated inside the container. However, the `SDK_NATIVE_SYSROOT` variable only exists in the container, not on my local computer.

This behavior seems counterintuitive, as the settings in `devcontainer.json` are meant for the container, not my local machine. Is there a way to define `cmake.cmakePath` exclusively for the container without using `.vscode/settings.json`? Using the latter would also override settings for users working locally.

According to [this](https://github.com/devcontainers/spec/issues/485) issue we should use `${containerEnv:SDK_NATIVE_SYSROOT}`, however this is not supported by this extension.

**Comment by gcampbell-msft** [comment]
@skycaptain If this issue is still bothering you, it definitely seems like a reasonable request that we could investigate. Without investigation I can't say with 100% confidence that we'll be able to implement your suggestion, but please create an issue for it and we'll track it there and people can upvote it if they'd also like that feature. 

Thanks!

**Reply by skycaptain** [reply]
Thanks for having a look. For the moment we kind of mitigated the issue by adding `${SDK_NATIVE_SYSROOT}/usr/bin` to `PATH`, i.e.

```json
{
  "image": "[redacted]",
  "remoteEnv": {
    "PATH": "${containerEnv:SDK_NATIVE_SYSROOT}/usr/bin:${containerEnv:PATH}"
  },
  "customizations": {
    "vscode": {
      "extensions": [
        "ms-vscode.cpptools-extension-pack"
      ],
      "settings": {
        "cmake.cmakePath": "cmake"
      }
    }
  }
}
```

However, I must say that having the native SDK executables permanently in the PATH has both advantages and disadvantages. I will create an issue and we'll see what others think.
